### DISCUSSION TITLE
How do I test the INVO application?

### DISCUSSION THREAD

**Discussion by s-ohnishi** [phalcon/cphalcon#16218] [Show and tell]
How do I test my INVO application?

I forked phalcon/invo into my repository.
And after cloning, I'm trying to test it in my local environment.
However, in step 4 of Installation in README.md
> Run DB migrations `vendor/bin/phalcon-migrations migration run --config=migrations.php`

cannot run.
That's because phalcon-migrations doesn't exist.
There is no such description in composer.json either.
There is migrations.php in the project root, but it uses Phalcon\Config inside, so it can be estimated that it is v4.
From this, it is highly likely that README.md is incorrect.

Also, there are .dat and .php files under db\migrations\1.0.0, but they use Phalcon\Migrations\Mvc\Model\Migration, which is also not available in v5.
(INVO is v.2, so it's a natural result)

I would like to make a proposal to modify the description so that it is compatible with Phalcon v.5, as it will also be my own learning, but I do not understand the migration description, so the only way to test it is to write directly to the DB using SQL. .
Please show me a sample migration script.

**Comment by tuannguyen29** [comment]
About the first error, you have to run the command `composer install` for it to download the dependencies. and to run the above command you must install Composer first.

As for the command `vendor/bin/phalcon-migrations migration run --config=migrations.php` on README.md is also wrong.
correct command for current version:
1. Show list `vendor/bin/phalcon-migrations list --config=migrations.php`
2. Generate `vendor/bin/phalcon-migrations generate --config=migrations.php`
3. Run create db `vendor/bin/phalcon-migrations run --config=migrations.php`

However, you must be able to connect to the database, adjust the configs in the `migrations.php` file to match your environment.

**Reply by tuannguyen29** [reply]
additionally, if `composer.json` does not have `"phalcon/migrations": "^3.0"` then run this command `composer require --dev phalcon/migrations`
see this repo: https://github.com/phalcon/migrations

**Reply by s-ohnishi** [reply]
Thank you for your reply.
I will try your suggested method.

**Reply by s-ohnishi** [reply]
There were many things I didn't understand, and it took a lot of time. Please forgive me.
When I tried the method you taught me, there were some parts that had to be modified, but the migration was successful.
(However, I'm not sure about the legitimacy of the fix.)

Thank you very much.

Below are the details of the modified parts:
https://github.com/phalcon/invo/issues/101
https://github.com/phalcon/invo/issues/102

**Reply by drowsyrearend** [reply]
> additionally, if `composer.json` does not have `"phalcon/migrations": "^3.0"` then run this command `composer require --dev phalcon/migrations` see this repo: 
[https://github.com/phalcon/](https://github.com/phalcon/migrations)<a href="https://retro-goal.com/">retro goal</a>
nice to know that
